src/common/net/fetch-client.ts

```diff
@@ -151,6 +200,93 @@ export class FetchHttpClient extends HttpClient implements HttpClientInterface {
 
     return new FetchHttpClientResponse(res);
   }
+
+  private async fetchRequestWithRetry(
```

Comment:
@stanleyphu, is there any way we could wrap the `fetchRequest` method above to avoid having to duplicate most of its logic? For example, I wonder if the section of code inside the `try` block could call `fetchRequest`